### PR TITLE
Support devices in LinuxParameters

### PR DESCRIPTION
**Pull request by ayemos**
https://github.com/eagletmt/hako/pull/37 
^ similar background 

ECS now supports `devices` attribute in  LinuxParameters option where we can pass `--device`  option to docker run .

@eagletmt please review 🙏

### CODE REVIEW

File: lib/hako/schedulers/ecs.rb [hunk 1]
```diff
@@ -1007,6 +1007,20 @@ def print_definition_in_cli_format(definition, additional_env: {})
...
           end
+
+          if definition[:linux_parameters][:devices]
+            devs = definition[:linux_parameters][:devices]
+            devs.each do |dev|
+              opts = dev[:host_path]
+              opts += ":#{dev[:host_path]}" if dev[:host_path]
+              if dev[:permissions]
+                dev[:permissions].each do |permission|
+                  opts += permission[0] if ['read', 'write', 'mknod'].include?(permission)
```
Comment (eagletmt) [inline lib/hako/schedulers/ecs.rb]: Use `%w[read write mknod]` style as Rubocop suggests

File: lib/hako/container.rb [hunk 2]
```diff
@@ -115,16 +115,28 @@ def extra_hosts
...
         conf = @definition['linux_parameters']
+
         if conf.key?('capabilities')
           cap = conf['capabilities']
-          {
-            capabilities: {
-              add: cap.fetch('add', []),
-              drop: cap.fetch('drop', [])
-            }
+          ret[:capabilities] = {
```
Comment (eagletmt) [inline lib/hako/container.rb]: `ret` seems to be initialized with `nil`

File: lib/hako/schedulers/ecs.rb [hunk 3]
```diff
@@ -1007,6 +1007,20 @@ def print_definition_in_cli_format(definition, additional_env: {})
               end
             end
           end
+
+          if definition[:linux_parameters][:devices]
+            devs = definition[:linux_parameters][:devices]
+            devs.each do |dev|
+              opts = dev[:host_path]
+              opts += ":#{dev[:host_path]}" if dev[:host_path]
```
Comment (eagletmt) [inline lib/hako/schedulers/ecs.rb]: `dev[:container_path]` ?

**Comment by ayemos** [comment]
@eagletmt please review 🙏

**Comment by eagletmt** [comment]
That depends on what is returned from DescribeTaskDefinition.
It seems that DescribeTaskDefinition returns linux_parameters=nil when no linux_parameters is configured. So this pull-request looks good.

**Comment by ayemos** [comment]
minor release together with https://github.com/eagletmt/hako/commit/45891a456a7d4bb96dffcf4de2df1ac5bd11c457 ?

**Comment by eagletmt** [comment]
Released v1.8.4 💎 
https://rubygems.org/gems/hako/versions/1.8.4

**Pull request merged**